Automattic/wp-calypso
# Plans: Remove english locale check for plans "Compare all plans" button label

Related to https://github.com/Automattic/wp-calypso/pull/91725

## Proposed Changes

* Remove english locale check for plans "Compare all plans" button labe

## Why are these changes being made?


* Allow all locales to see the label change

## Testing Instructions



- Checkout branch or use live link
- Navigate to /start/guided and go through the flow until you reach the plans page.
- Guided onboarding flow plan page: Confirm the Compare all plans label appears correctly and the button functions as expected.
- Navigate to /start/ and go through the flow until you reach the plans page.
- The plan comparison button label should display Compare plans
- Change locale and verify the text is translated

## Pre-merge Checklist
